File: src/1750-Minimum-Length-of-String-After-Deleting-Similar-Ends/minimum_length_of_string_after_deleting_similar_ends.py
class Solution:
    def minimumLength(self, s: str) -> int:
        left: int = 0
        right: int = len(s) - 1

        if len(s) == 1:
            return 1
        
        while left < right and s[left] == s[right]:
            c: str = s[left]
            while left <= right and s[left] == c:
                left += 1
            while right >= left and s[right] == c:
                right -= 1
        
        ans: int = right - left + 1
        return ans
            

if __name__ == "__main__":
    s: str = "bbbbbbbbbbbbbbbbbbbbbbbbbbbabbbbbbbbbbbbbbbccbcbcbccbbabbb"
    solution = Solution()

    print(solution.minimumLength(s))

File: src/1750-Minimum-Length-of-String-After-Deleting-Similar-Ends/minimum_length_of_string_after_deleting_similar_ends.cpp
#include <iostream>
#include <string>

class Solution {
public:
    int minimumLength(std::string s) {
        int left = 0;
        int right = s.size() - 1;

        if (s.size() == 1) return 1;

        while (left < right && s[left] == s[right]) {
            char c = s[left];

            while (left <= right && s[left] == c) {
                left += 1;
            }
            while (right >= left && s[right] == c) {
                right -= 1;
            }
        }

        int ans = right - left + 1;
        return ans;
    }
};

int main() {
    std::string s = "bbbbbbbbbbbbbbbbbbbbbbbbbbbabbbbbbbbbbbbbbbccbcbcbccbbabbb";
    Solution solution;

    std::cout << solution.minimumLength(s);

    return 0;
}